.test-defs/cmd/shoot-maintenance/main.go

```diff
@@ -57,11 +60,87 @@ var (
 	testMachineImage = gardenv1beta1.ShootMachineImage{
 		Version: testMachineImageVersion,
 	}
+
+	shootMaintenanceTest          *framework.ShootMaintenanceTest
+	gardenerSchedulerReplicaCount *int32
 )
 
 func init() {
+	ctx := context.Background()
+	defer ctx.Done()
 	testLogger = logger.NewLogger("debug")
 
+	setEnvironmentVariables()
+	setupShootMaintenanceTest(ctx)
+	setupEnvironmentForMaintenanceTest()
+}
+
+func setupShootMaintenanceTest(ctx context.Context) {
+	gardenerConfigPath := fmt.Sprintf("%s/gardener.config", kubeconfigPath)
+	shootGardenerTest, err := framework.NewShootGardenerTest(gardenerConfigPath, nil, testLogger)
+	if err != nil {
+		testLogger.Fatalf("Cannot create ShootGardenerTest %s", err.Error())
```

Comment:
You could return the `err`s from this func or fail gracefully. The current way is just log the err and ignore it which is not expected I guess.

Reply:
logger.FatalF is logging the error and then  shuts down the process (defaults to os.Exit()).
I think this is the expected behaviour as it does not make sense to continue execution at this point.

Reply:
Yep, you are right. My bad.